clang/include/clang/ExtractAPI/ExtractAPIVisitor.h

```diff
@@ -1146,11 +1146,30 @@ bool ExtractAPIVisitorBase<Derived>::VisitTypedefNameDecl(
 
   StringRef Name = Decl->getName();
 
+  auto nameMatches = [&Name](TagDecl *TagDecl) {
+    StringRef TagName = TagDecl->getName();
+
+    if (TagName == Name)
+      return true;
+
+    // Also check whether the tag decl's name is the same as the typedef name
+    // with prefixed underscores
+    if (TagName.starts_with('_')) {
+      StringRef StrippedName =
+          TagName.drop_while([](char c) { return c == '_'; });
```

Comment:
nit: maybe a bit simpler to use `ltrim('_')`

Reply:
I knew i was missing a simpler option! Thanks for the pointer; i've pushed up a new commit that swaps it out.